### PR TITLE
[RTM] ENH: Add --low-mem option to not compress resampled BOLD series

### LINKED ISSUE #657: MemoryError in SignalExtraction
```
170809-20:04:16,30 workflow INFO:
	 Executing node signals in dir: /scratch/fmriprep_wf/single_subject_03_wf/func_preproc_task_mapper_run_02_wf/bold_confounds_wf/signals
170809-20:05:06,645 workflow ERROR:
	 ['Node signals failed to run on host 2fae67b907ed.']
170809-20:05:06,778 workflow INFO:
	 Saving crash info to /out/fmriprep/sub-03/log/20170809-192806_05fb6cec-029a-492e-9ec1-2a9d20efb32b/crash-20170809-200506-root-signals-2ea82a32-de8b-4ab4-990e-5f40755db365.txt
170809-20:05:06,779 workflow INFO:
	 Traceback (most recent call last):
  File "/usr/local/miniconda/lib/python3.6/site-packages/niworkflows/nipype/pipeline/plugins/multiproc.py", line 52, in run_node
    result['result'] = node.run(updatehash=updatehash)
  File "/usr/local/miniconda/lib/python3.6/site-packages/niworkflows/nipype/pipeline/engine/nodes.py", line 372, in run
    self._run_interface()
  File "/usr/local/miniconda/lib/python3.6/site-packages/niworkflows/nipype/pipeline/engine/nodes.py", line 482, in _run_interface
    self._result = self._run_command(execute)
  File "/usr/local/miniconda/lib/python3.6/site-packages/niworkflows/nipype/pipeline/engine/nodes.py", line 613, in _run_command
    result = self._interface.run()
  File "/usr/local/miniconda/lib/python3.6/site-packages/niworkflows/nipype/interfaces/base.py", line 1083, in run
    runtime = self._run_wrapper(runtime)
  File "/usr/local/miniconda/lib/python3.6/site-packages/niworkflows/nipype/interfaces/base.py", line 1031, in _run_wrapper
    runtime = self._run_interface(runtime)
  File "/usr/local/miniconda/lib/python3.6/site-packages/niworkflows/nipype/interfaces/nilearn.py", line 83, in _run_interface
    signals.append(masker.fit_transform(self.inputs.in_file))
  File "/usr/local/miniconda/lib/python3.6/site-packages/nilearn/input_data/nifti_maps_masker.py", line 214, in fit_transform
    return self.fit().transform(imgs, confounds=confounds)
  File "/usr/local/miniconda/lib/python3.6/site-packages/nilearn/input_data/base_masker.py", line 176, in transform
    return self.transform_single_imgs(imgs, confounds)
  File "/usr/local/miniconda/lib/python3.6/site-packages/nilearn/input_data/nifti_maps_masker.py", line 254, in transform_single_imgs
    imgs_ = _utils.check_niimg_4d(imgs)
  File "/usr/local/miniconda/lib/python3.6/site-packages/nilearn/_utils/niimg_conversions.py", line 361, in check_niimg_4d
    dtype=dtype)
  File "/usr/local/miniconda/lib/python3.6/site-packages/nilearn/_utils/niimg_conversions.py", line 271, in check_niimg
    niimg = load_niimg(niimg, dtype=dtype)
  File "/usr/local/miniconda/lib/python3.6/site-packages/nilearn/_utils/niimg.py", line 116, in load_niimg
    dtype = _get_target_dtype(niimg.get_data().dtype, dtype)
  File "/usr/local/miniconda/lib/python3.6/site-packages/nibabel/spatialimages.py", line 572, in get_data
    data = np.asanyarray(self._dataobj)
  File "/usr/local/miniconda/lib/python3.6/site-packages/numpy/core/numeric.py", line 583, in asanyarray
    return array(a, dtype, copy=False, order=order, subok=True)
  File "/usr/local/miniconda/lib/python3.6/site-packages/nibabel/arrayproxy.py", line 145, in __array__
    raw_data = self.get_unscaled()
  File "/usr/local/miniconda/lib/python3.6/site-packages/nibabel/arrayproxy.py", line 140, in get_unscaled
    mmap=self._mmap)
  File "/usr/local/miniconda/lib/python3.6/site-packages/nibabel/volumeutils.py", line 523, in array_from_file
    data_bytes = bytearray(n_bytes)
MemoryError: 
Interface SignalExtraction failed to run.
```

While looking into #652, I also ran across this error. On a 16GB system w/ 16GB swap, memory and most (~13GB) of swap filled before failure. Had closed all large apps before running.

### PR DESCRIPTION
**Pull request by effigies**
Closes #657.

### CODE REVIEW

File: fmriprep/interfaces/bids.py [hunk 1]
```diff
@@ -207,7 +211,12 @@ def _run_interface(self, runtime):
             if isdefined(self.inputs.extra_values):
                 out_file = out_file.format(extra_value=self.inputs.extra_values[i])
             self._results['out_file'].append(out_file)
-            copy(self.inputs.in_file[i], out_file)
+            if compress:
+                with open(fname, 'rb') as f_in:
```
Comment (oesteban) [inline fmriprep/interfaces/bids.py]: We are re-setting `fname` here. We shouldn't be doing that as it is confusing

File: fmriprep/interfaces/bids.py [hunk 2]
```diff
@@ -166,6 +167,9 @@ def __init__(self, out_path_base=None, **inputs):
     def _run_interface(self, runtime):
         fname, _ = _splitext(self.inputs.source_file)
         _, ext = _splitext(self.inputs.in_file[0])
+        compress = ext == '.nii'
```
Comment (oesteban) [inline fmriprep/interfaces/bids.py]: This will compress the derivatives always. Is that expected?

Comment (effigies) [inline fmriprep/interfaces/bids.py]: Yes. I don't think we currently have an option for uncompressed `.nii` outputs.

**Comment by oesteban** [comment]
Cool, just double checking

File: fmriprep/interfaces/bids.py (hunk 1 above)
Comment (effigies) [inline fmriprep/interfaces/bids.py]: I can fix that.

Comment (oesteban) [inline fmriprep/interfaces/bids.py]: ups, I forgot about this

Comment (effigies) [inline fmriprep/interfaces/bids.py]: Ah well. I'll make a quick PR.

**Pull request merged**